### PR TITLE
SUS-1182 | SpecialCreateNewWiki: check if user has confirmed his email	

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-1182

### CODE REVIEW

File: extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php [hunk 1]
```diff
@@ -6,25 +6,30 @@ public function __construct() {
...
-			$this->displayRestrictionError();
-			wfProfileOut( __METHOD__ );
-			return;
+		// SUS-1182: CreateNewWiki should check for valid email before progressing to the second step
+		if ( !$this->getUser()->isEmailConfirmed() ) {
+			throw new ErrorPageError( 'cnw-error-unconfirmed-email-header', 'cnw-error-unconfirmed-email' );
 		}
 
-		$wgOut->setPageTitle(wfMsg('cnw-title'));
+		$out->setPageTitle(wfMsg('cnw-title'));
```
Comment (WikiaTech) [inline extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php]: MAJOR Replace this legacy wfMsg* function call with one of the new i18n functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php [hunk 2]
```diff
@@ -3,28 +3,33 @@
...
-			$this->displayRestrictionError();
-			wfProfileOut( __METHOD__ );
-			return;
+		// SUS-1182: CreateNewWiki should check for valid email before progressing to the second step
+		if ( !$this->getUser()->isEmailConfirmed() ) {
+			throw new ErrorPageError( 'cnw-error-unconfirmed-email-header', 'cnw-error-unconfirmed-email' );
 		}
 
-		$wgOut->setPageTitle(wfMsg('cnw-title'));
+		$out->setPageTitle( wfMessage( 'cnw-title' ) );
```
Comment (mszabo-wikia) [inline extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php:32]: We could actually use `$this->msg()` here.

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php (hunk 2 above)
Comment (macbre) [reply extensions/wikia/CreateNewWiki/SpecialCreateNewWiki.class.php:32]: `setPageTitle` method allows `Message` instance to be passed.

**Pull request merged**